rust-connectors/sources/http/src/formatter.rs

```diff
@@ -0,0 +1,43 @@
+// Pre-Allocations ::with_capacity()
+#![allow(clippy::vec_init_then_push)]
+
+/// Assembles and allocates the final full Record string
+pub fn format_full_record(
+    version: &str,
+    status: &str,
+    header_count: usize,
+    headers: &str,
+    body: &str,
+) -> String {
+    let mut record_out_parts: Vec<String> = Vec::with_capacity(4);
+    let mut status_line: Vec<String> = Vec::with_capacity(2);
+    status_line.push(version.to_owned());
+    status_line.push(status.to_owned());
+    record_out_parts.push(status_line.join(" "));
+    if header_count > 0 {
+        record_out_parts.push(headers.to_owned());
+    }
+    record_out_parts.push(String::from(""));
+    record_out_parts.push(body.to_owned());
+    record_out_parts.join("\n")
+}
+
+/// Format headers into full response
+pub fn format_reqwest_headers(hdr_map: &reqwest::header::HeaderMap) -> String {
+    let mut hdr_vec = Vec::with_capacity(hdr_map.len());
```

Comment:
I think that this should be `Vec::with_capacity(hdr_map.len() + 1);`

Reply:
Can't see where the extra item comes?

Reply:
Nevermind, I thought that the last line was `.push` instead of `.join`